htmd/parameterization/util.py

```diff
@@ -383,6 +383,72 @@ def makeAtomNamesUnique(mol):
 
     return mol
 
+def detectChiralCenters(mol):
+    """
+    Detect chiral centers
+
+    Parameters
+    ----------
+    mol: Molecule
+        Molecule to detect chiral centers
+
+    Return
+    ------
+    results: list of list of tuples
+        List of list of chircal centers for each frame.
+        The chiral centers are tuples made of an atom index and a label ('S', 'R', '?').
+
+    Examples
+    --------
+    >>> from htmd.home import home
+    >>> from htmd.molecule.molecule import Molecule
+
+    >>> molFile = os.path.join(home('test-param'), 'H2O2.mol2')
+    >>> mol = Molecule(molFile)
+    >>> detectChiralCenters(mol)
+    [[]]
+
+    >>> molFile = os.path.join(home('test-param'), 'fluorchlorcyclopronol.mol2')
+    >>> mol = Molecule(molFile)
+    >>> detectChiralCenters(mol)
+    [[(0, 'R'), (2, 'S'), (4, 'R')]]
+
+    >>> mol.appendFrames(mol)
+    >>> detectChiralCenters(mol)
+    [[(0, 'R'), (2, 'S'), (4, 'R')], [(0, 'R'), (2, 'S'), (4, 'R')]]
+    """
+
+    from htmd.molecule.molecule import Molecule
+    from rdkit.Chem import MolFromMol2File, AssignAtomChiralTagsFromStructure, FindMolChiralCenters
+
+    if not isinstance(mol, Molecule):
+        raise TypeError('"mol" has to be instance of {}'.format(Molecule))
+
+    all_chiral_centers = []
+
```

Comment:
Copy mol once here, set atomtypes once. `htmd_mol = mol.copy()  htmd_mol.atomtype = htmd_mol.element`
In the loop just change the `htmd_mol.frame = i` variable and it will write the correct frame out